interactive_engine/executor/ir/core/src/plan/logical.rs

```diff
@@ -1272,6 +1326,71 @@ impl AsLogical for pb::IndexPredicate {
     }
 }
 
+fn is_whole_graph(operator: &pb::logical_plan::Operator) -> bool {
+    if let Some(opr) = &operator.opr {
+        match opr {
+            pb::logical_plan::operator::Opr::Scan(scan) => {
+                scan.idx_predicate.is_none()
+                    && scan.alias.is_none()
+                    && scan
+                        .params
+                        .as_ref()
+                        .map(|params| !params.is_queryable() && is_params_all_labels(params))
+                        .unwrap_or(true)
+            }
+            pb::logical_plan::operator::Opr::Root(_) => true,
+            _ => false,
+        }
+    } else {
+        false
+    }
+}
+
+fn is_params_all_labels(params: &pb::QueryParams) -> bool {
+    if let Ok(store_meta) = STORE_META.read() {
+        if let Some(schema) = store_meta.schema.as_ref() {
+            params.tables.is_empty() || {
+                let params_label_ids: BTreeSet<LabelId> = params
+                    .tables
+                    .iter()
+                    .filter_map(|name_or_id| {
+                        if let Some(ir_common::generated::common::name_or_id::Item::Id(id)) =
+                            name_or_id.item.as_ref()
+                        {
+                            Some(*id)
+                        } else {
+                            None
+                        }
+                    })
+                    .collect();
+                let meta_label_ids: BTreeSet<LabelId> = schema
+                    .entity_labels_iter()
+                    .map(|(_, label_id)| label_id)
+                    .collect();
+                params_label_ids.eq(&meta_label_ids)
+            }
+        } else {
+            false
+        }
+    } else {
+        false
```

Comment:
Even if store_meta is not provided, params.tables.is_empty() seems also indicates all_labels()?

Reply:
Fixed